x-pack/plugins/maps/public/classes/layers/create_tile_map_layer_descriptor.ts

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  AggDescriptor,
+  ColorDynamicOptions,
+  LayerDescriptor,
+  SizeDynamicOptions,
+  VectorStylePropertiesDescriptor,
+} from '../../../common/descriptor_types';
+import {
+  AGG_TYPE,
+  COLOR_MAP_TYPE,
+  FIELD_ORIGIN,
+  GRID_RESOLUTION,
+  RENDER_AS,
+  STYLE_TYPE,
+  VECTOR_STYLES,
+} from '../../../common/constants';
+import { VectorStyle } from '../styles/vector/vector_style';
+// @ts-ignore
+import { ESGeoGridSource } from '../sources/es_geo_grid_source';
+import { VectorLayer } from './vector_layer/vector_layer';
+// @ts-ignore
+import { HeatmapLayer } from './heatmap_layer/heatmap_layer';
+import { getDefaultDynamicProperties } from '../styles/vector/vector_style_defaults';
+import { NUMERICAL_COLOR_PALETTES } from '../styles/color_palettes';
+import { getSourceAggKey } from '../../../common/get_agg_key';
+import { isMetricCountable } from '../util/is_metric_countable';
+
+const defaultDynamicProperties = getDefaultDynamicProperties();
+
+function isHeatmap(mapType: string): boolean {
+  return mapType.toLowerCase() === 'heatmap';
+}
+
+function getGeoGridRequestType(mapType: string): RENDER_AS {
+  if (isHeatmap(mapType)) {
+    return RENDER_AS.HEATMAP;
+  }
+
+  if (mapType.toLowerCase() === 'shaded geohash grid') {
+    return RENDER_AS.GRID;
+  }
+
+  return RENDER_AS.POINT;
+}
+
+export function createAggDescriptor(
+  mapType: string,
+  metricAgg: string,
+  metricFieldName?: string
+): AggDescriptor {
+  const aggTypeKey = Object.keys(AGG_TYPE).find((key) => {
+    return AGG_TYPE[key as keyof typeof AGG_TYPE] === metricAgg;
+  });
+  const aggType = aggTypeKey ? AGG_TYPE[aggTypeKey as keyof typeof AGG_TYPE] : undefined;
+
```

Comment:
This would change the flow somewhat, but form here we are creating maps that might show metrics we don't support. 

e.g. for pipeline aggrevations

Maybe we should just _not_ show the "view in maps" button when this is the case?

Reply:
fwiw, we'd have similar issues with e.g. users setting advanced0json (like min_doc_count).

Reply:
> Maybe we should just not show the "view in maps" button when this is the case?

These users will still need to migrate to maps in 8.0 so I think its best to send them to maps with at least a map showing count.